### PR TITLE
chore: Refactored client channels to React

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR introduces a refactoring of the channels and its partials to React.
- All Nunjucks partials are reworked to components
- File template are reworked to React

### CODE REVIEW

File: __react/components/channel/request.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+
+  //Determine the request operation based on whether the message type is null
+  let requestOperation = `msg = await nc.request(${realizeChannelName(channelParameters, channelName)}, timeout, null)`
+  if(messageHasNotNullPayload(requestMessage.payload())){
+    requestOperation = `
+    ${OnSendingData(requestMessage, defaultContentType)}
+    msg = await nc.request(${realizeChannelName(channelParameters, channelName)}, timeout, dataToSend)
+    `;
+  }
```
Comment (magicmatatjahu) [inline __react/components/channel/request.js:31]:
```ts
  let requestOperation = "";
  if(messageHasNotNullPayload(requestMessage.payload())){
    requestOperation = `
    ${OnSendingData(requestMessage, defaultContentType)}
    msg = await nc.request(${realizeChannelName(channelParameters, channelName)}, timeout, dataToSend)
    `;
  } else {
    requestOperation = `msg = await nc.request(${realizeChannelName(channelParameters, channelName)}, timeout, null)`
  }
```

should be better :) Also in another places. But this is only a suggestion.

File: __react/template/src/channels/$$channel$$.ts.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        channel.parameters());
+    }
+  }
+  return channelcode;
+}
+
+export default function channelRender({ asyncapi, channelName, channel, params }) {
+  // Import the correct messages
+  let publishMessageImport = '';
+  if(channel.hasPublish() && messageHasNotNullPayload(channel.publish().message(0).payload())){
```
Comment (magicmatatjahu) [inline __react/template/src/channels/$$channel$$.ts.js:62]: This same situation with redundant calling some parser functions. `const msg = channel.publish().message(0)` should speed the code :) Here in another places in this function.

**Review by magicmatatjahu — COMMENTED** [review]
Good! Some suggestions :)

File: __react/components/channel/ChannelParameterUnwrap.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    if (index === 0) {
+      channelSplit = `channel = channel.substring(splits[${index}].length);`;
+    }
+    prevParameterName = camelCase(parameterName);
+    const paramVarToCast = `channel.substring(0, ${prevParameterName}End)`;
+    return `
+			${channelSplit}
+			var ${prevParameterName}End = channel.indexOf(splits[${index+1}]);
+			var ${prevParameterName}Param = ${castToTsType(parameter.schema().type(), paramVarToCast)};
+		`;
```
[the remark this answers is not in the document]
Comment (jonaslagoni) [reply __react/components/channel/ChannelParameterUnwrap.js:75]: There was a mix of tabs and spaces, fixed 😄

File: __react/components/channel/ChannelParameterUnwrap.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  //Create the split array which contains the string between each parameter
+  let splits = '';
+  splits = Object.entries(channelParameters).map(([parameterName, _], index) => {
+    const parameterCamelCase = camelCase(parameterName);
+    // Check if we reached the end of the parameter list
+    if (index+1 === Object.keys(channelParameters).length) {
+      return `
+			${parameterCamelCase}Split[0],
+			${parameterCamelCase}Split[1]
+			`;
```
[the remark this answers is not in the document]
Comment (jonaslagoni) [reply __react/components/channel/ChannelParameterUnwrap.js:55]: There was a mix of tabs and spaces, fixed 😄

File: __react/components/channel/OnReceivingData.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    convertToString = `
+    if(receivedDataHooks.length == 0){
+      receivedData = ${pascalCase(message.uid())}Message.Convert.to${pascalCase(message.uid())}(receivedData);
+    }`;
+  }
+
+  return `
+  try {
+    let receivedDataHooks = Hooks.getInstance().getreceivedDataHook();
+    var receivedData : any = msg.data;
```
[the remark this answers is not in the document]
Comment (jonaslagoni) [reply __react/components/channel/OnReceivingData.js]: Yea will have to keep that in mind 👍

File: __react/components/channel/request.js (hunk 1 above)
Comment (jonaslagoni) [reply __react/components/channel/request.js:32]: Do you feel like this provides a clearer understanding of the code right of the bat with your suggestion? @magicmatatjahu

File: __react/template/src/channels/$$channel$$.ts.js (hunk 2 above)
Comment (jonaslagoni) [reply __react/template/src/channels/$$channel$$.ts.js]: @magicmatatjahu would you always create a variable for a value if used more then once?

Comment (magicmatatjahu) [reply __react/template/src/channels/$$channel$$.ts.js]: In some situation not, but here, calling `channel.publish().message(0)` we first create the `Publish` class's instance and then we create the `Message` class's instance, so I have problem with optimization :)

File: __react/components/channel/request.js (hunk 1 above)
Comment (magicmatatjahu) [reply __react/components/channel/request.js:32]: I think that it's more verbose, but it's only a suggestion :)

Comment (jonaslagoni) [reply __react/components/channel/request.js:32]: Think I will leave it as is for now 😄

File: __react/template/src/channels/$$channel$$.ts.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  return <File name={`${pascalCase(channelName)}.ts`}>
+    {`
+    ${publishMessageImport}
+    ${subscribeMessageImport}
+
+    import { Client, NatsError, Subscription, SubscriptionOptions, Payload } from 'ts-nats';
+    import {ErrorCode, NatsTypescriptTemplateError} from '../NatsTypescriptTemplateError';
+    import { Hooks } from '../hooks';
+
+    ${getChannelCode(asyncapi, channel, channelName, params)}
```
Comment (magicmatatjahu) [inline __react/template/src/channels/$$channel$$.ts.js:86]: This block will be also intend in output file :)

**Review by magicmatatjahu — APPROVED** [review]

File: __react/template/src/channels/$$channel$$.ts.js (hunk 6 above)
Comment (jonaslagoni) [reply __react/template/src/channels/$$channel$$.ts.js:86]: I will make an adjustment in the issue with #20

**Comment by jonaslagoni** [comment]
Thanks for the review @magicmatatjahu ! 🙇

**Pull request merged**